pkgs/applications/networking/instant-messengers/signal-desktop/default.nix

```diff
@@ -10,4 +10,16 @@
       platforms = [ "x86_64-linux" ];
     };
   });
+  signal-desktop-apple-emoji =
+    if hostPlatform.system == "aarch64-linux" then
+      callPackage ./signal-desktop-aarch64_apple-emoji.nix { }
+    else
+      callPackage ./signal-desktop_apple-emoji.nix { };
```

Comment:
I think the filename should match the package name here.

However, I personally think it would be better to support the standard `.override` interface here rather than exposing aliases. To solve the issue with that you brought up, I think you could perhaps instead do something like `signal-desktop = callPackage ({ withUnfree ? false }: if withUnfree then callPackage … else callPackage …)`? I’m not sure how that would work in terms of overriding `signal-desktop`’s dependencies, though.

Reply:
> I think the filename should match the package name here.

Changed.

> However, I personally think it would be better to support the standard `.override` interface here rather than exposing aliases. 

I agree, but I don't see how to get `nix-update` to update `signal-desktop.override { withUnfree = true; }` since it only takes attributes. Your tip allows for overriding, but I'm not sure how it addresses this.

> To solve the issue with that you brought up, I think you could perhaps instead do something like `signal-desktop = callPackage ({ withUnfree ? false }: if withUnfree then callPackage … else callPackage …)`? I’m not sure how that would work in terms of overriding `signal-desktop`’s dependencies, though.

Implemented this, but I still have the aliases solely for `nix-update` to find it. This is a bit awkward I agree.

Reply:
Hmm, the point about `nix-update` is a good one. It should really take an expression. cc @Mic92 any suggestions for how to handle cases where package feature flags need separate updates?

I won’t block this on that since it seems tricky to solve, and good point about the overrides. Would still like to hear feedback from other Signal maintainers though (cc @teutat3s?).

(I wish we built Signal from source so that it wasn’t so awkward to update and patch…)